.github/CONTRIBUTING.md

```diff
@@ -1,32 +1,122 @@
+check if you are using the latest version of the project,
+[found here](https://github.com/checkstyle/checkstyle/releases). If you are not up-to-date,
+see if updating to the latest release fixes your issue.
 
-* [Importing checkstyle project to Netbeans](https://checkstyle.org/netbeans.html)
+A great way to contribute to the project is to create a detailed issue when you encounter
+a problem or would like to suggest a feature. We always appreciate a well-written,
+thorough issue description. :brain:
+
+Some points to consider when opening an issue:
+- **Fully complete the provided issue template.** The bug report and feature request templates
+    specify all the information we need to quickly and efficiently address your issue. Be clear,
+    concise, and descriptive.
+    Provide as much information as you can, including steps to reproduce, stack traces, etc.
+- **Use [GitHub-flavored Markdown](https://help.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax).**
+    Especially put code blocks and console outputs in backticks (```). This improves readability.
+- **Do not open a duplicate feature request.** Search for existing feature requests first. If you
+    find your feature (or one very similar) previously requested, comment on that issue.
+
+## :lock: Reporting Security Issues
+
+**Do not file a public issue for security vulnerabilities.** Please contact the 
+maintainers directly.
+see the [Security Policy](https://github.com/checkstyle/SECURITY.md) for more information.
+
+## :technologist: Submitting Pull Requests
+
+ - **Read our pull request rules.** See [PR Rules](https://github.com/checkstyle/checkstyle/wiki/PR-rules).
+ - **Comment on the issue.** If you are working on an existing issue, please comment on the issue
+    to let others know that you are working on it ("I am on it.").
+ - **Read the Github docs.** Visit GitHub's [Pull Request Guide](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests)
+    for information on how to submit a pull request.
+ - **Follow the template.** Please follow the [CheckStyle Pull Request Template](https://github.com/checkstyle/checkstyle/blob/master/.github/PULL_REQUEST_TEMPLATE.md)
+    that is provided in the pull request description when submitting a pull request.
+ - **Run maven build locally.** `mvn clean verify` should pass on your local before
+   submitting a pull request.
+
+## :white_check_mark: Code Review
+
+All submissions, including submissions by project members, require review. We use GitHub pull
+requests for this purpose. Consult the [GitHub Help](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-reviews)
+for more information on pull request reviews.
+
+Here are some general guidelines to follow when submitting a pull request:
+ - **Reply to comments.** If a reviewer asks for changes, make the changes and reply to each
```

Comment:
> make the changes

I think this wording might be too strong. I have made requests and found out I was wrong when contributors explained why. There should be a conversation if the changes cannot be made for a reason.

Also I think this paragraph might be better after open to feedback as it flows with the purpose of replying and updating.

Reply:
I agree and already felt the same, fixed